src/metabase/public_settings.clj

```diff
@@ -205,38 +217,67 @@
       (tru "Invalid Setting: {0}/{1}" ns-symb setting-symb))
     (f)))
 
-;; TODO - it seems like it would be a nice performance win to cache this a little bit
-(defn public-settings
-  "Return a simple map of key/value pairs which represent the public settings (`MetabaseBootstrap`) for the front-end
-   application."
-  []
-  {:admin_email           (admin-email)
-   :anon_tracking_enabled (anon-tracking-enabled)
-   :available_locales     (available-locales-with-names)
-   :custom_formatting     (custom-formatting)
-   :custom_geojson        (resolve-setting 'metabase.api.geojson 'custom-geojson)
-   :email_configured      (resolve-setting 'metabase.email 'email-configured?)
-   :embedding             (enable-embedding)
-   :enable_nested_queries (enable-nested-queries)
-   :enable_query_caching  (enable-query-caching)
-   :enable_xrays          (enable-xrays)
-   :engines               (driver.u/available-drivers-info)
-   :entities              (types/types->parents :entity/*)
-   :ga_code               "UA-60817802-1"
-   :google_auth_client_id (resolve-setting 'metabase.api.session 'google-auth-client-id)
-   :has_sample_dataset    (db/exists? 'Database, :is_sample true)
-   :hide_embed_branding   (metastore/hide-embed-branding?)
-   :ldap_configured       (resolve-setting 'metabase.integrations.ldap 'ldap-configured?)
-   :map_tile_server_url   (map-tile-server-url)
-   :metastore_url         metastore/store-url
-   :password_complexity   password/active-password-complexity
-   :premium_token         (metastore/premium-embedding-token)
-   :public_sharing        (enable-public-sharing)
-   :report_timezone       (resolve-setting 'metabase.driver 'report-timezone)
-   :setup_token           (resolve-setting 'metabase.setup 'token-value)
-   :site_name             (site-name)
-   :site_url              (site-url)
-   :timezone_short        (short-timezone-name (setting/get :report-timezone))
-   :timezones             common/timezones
-   :types                 (types/types->parents :type/*)
-   :version               config/mb-version-info})
+(defsetting available-locales
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (available-locales-with-names)))
+
+(defsetting email-configured
```

Comment:
All of these settings that have `(resolve-setting)` function calls are defined in other namespaces. So it's a bit silly to have a setting like `email-configured` that is just an alias for `email-configured?` in another namespace. I would prefer we take out this version entirely and use `metabase.email/email-configured?` directly. 

It seems like it would be pretty trivial to have the `properties` function change settings with Clojurey-names like `email-configured?` to JS ones. You can do this:

```clj
(reduce (partial apply str/replace) 'email-configured? {#"-" "_", #"\?" ""})
;; -> "email_configured"
```

Reply:
Also we'd have to write real dox for all those `TODO`s here so if you go with the approach above it would save time

Reply:
I'd actually rather not do the magic to convert settings names to be JS friendly, it's been more trouble that it's worth. If we want JS friendly names we can add functions to `metabase/lib/settings.js`. We actually already have some for these `*-configured` settings: https://github.com/metabase/metabase/pull/11239/files#diff-385d015e99193a9cb64386d7c88716b2R59

I'll go ahead and convert `email-configured?` etc to a setting with `:visibility :public`. I think it should be something like this?

```
(defsetting email-configured?
  "Predicate function which returns `true` if we have a viable email configuration for the app, `false` otherwise."
  :internal?   true
  :visibility? :public
  :setter      :none
  :getter      (fn []  (boolean (email-smtp-host))))
```

Reply:
yeah, that will work. For tiny functions like that don't forget you can use the shorthand function syntax `#()` – you can do `#(boolean (email-smtp-host))` here instead.